sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -232,7 +234,7 @@ final class DataFrameWriter private[sql](df: DataFrame) {
    * @since 1.4.0
    */
   @scala.annotation.varargs
-  def partitionBy(colNames: String*): DataFrameWriter = {
+  def partitionBy(colNames: String*): this.type = {
```

Comment:
sorry i didnt notice this earlier but why is this suddenly `this.type` and not `DataFrameWriter[T]` like other methods. Alternatively, why are we using `this.type` everywhere?